### PR TITLE
feat: paid mortgages & claim

### PR DESCRIPTION
**Pull request by nachomazzara**

### CODE REVIEW

File: scripts/monitor/processEvents.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ async function processNoParcelRelatedEvents(event) {
         log.info(`[${name}] Cancelling Mortgage ${_id}`)
         await Mortgage.update(
           {
-            status: Mortgage.STATUS.cancelled,
```
Comment (NicoSantangelo) [inline scripts/monitor/processEvents.js]: While both `canceled` and `cancelled` are correct, the rest of the codebase uses `cancelled` for publication status. Can we change this one? just to be consistent

File: scripts/monitor/processEvents.js [hunk 2]
```diff
@@ -66,6 +67,105 @@ async function processNoParcelRelatedEvents(event) {
...
+      break
+    }
+    case BlockchainEvent.EVENTS.partialPayment: {
+      const { _index } = event.args
+      const block_time_updated_at = await new BlockTimestampService().getBlockTime(
+        block_number
+      )
+      log.info(`[${name}] Partial Payment for loan ${_index}`)
+
+      const rcnEngineContract = await eth.getContract('RCNEngine')
```
Comment (NicoSantangelo) [inline scripts/monitor/processEvents.js]: `getContract` is not `async`

File: scripts/monitor/processEvents.js [hunk 3]
```diff
@@ -66,6 +67,105 @@ async function processNoParcelRelatedEvents(event) {
...
+      try {
+        log.info(`[${name}] Starting Mortgage ${_id}`)
+        await Mortgage.update(
+          {
+            status: Mortgage.STATUS.ongoing,
+            block_time_updated_at
+          },
+          {
+            mortgage_id: _id
+          }
```
Comment (NicoSantangelo) [inline scripts/monitor/processEvents.js]: _Opinion_: I think that `{ mortgage_id: _id }` looks a bit better, same with `loan_id: _index` and the others below. If you agree and don't mind changing it, I'm all for it.

File: webapp/src/components/BuyParcelByMortgagePage/MortgageForm/MortgageForm.js [hunk 4]
```diff
@@ -288,33 +285,18 @@ export default class MortgageForm extends React.PureComponent {
...
-          <Message icon>
-            <Icon name="circle notched" loading />
-            <Message.Content>
-              <TxStatus.Text
-                txHash={publication.tx_hash}
-                txStatus={publication.tx_status}
-              />
-            </Message.Content>
-          </Message>
-        ) : null}
```
Comment (NicoSantangelo) [inline webapp/src/components/BuyParcelByMortgagePage/MortgageForm/MortgageForm.js]: Why are we deleting the pending Message here? I'm not necessarily opposed but I think it can be useful.

Also, upon further instruction, `isIdle` marks if the transaction started but hasn't been confirmed by the user. For example if you leave the metamask popup open. It's not that important, but might be useful for disallowing new txs until the last one is confirmed

File: webapp/src/components/MortgageActions/MortgageActions.container.js [hunk 5]
```diff
@@ -9,9 +16,18 @@ const mapState = (state, ownProps) => {
   }
 }
 
-const mapDispatch = (dispatch, { mortgage }) => ({
-  onCancel: () =>
-    dispatch(cancelMortgageRequest(mortgage.mortgage_id, mortgage.asset_id))
-})
+const mapDispatch = (dispatch, { mortgage }) => {
+  const coordinates = splitCoordinate(mortgage.asset_id)
```
Comment (NicoSantangelo) [inline webapp/src/components/MortgageActions/MortgageActions.container.js]: To be a bit more clear, we could change this to  `const [x, y] = splitCoordinate(asset_id)`

Also, looking at this I found a bug on `cancelMortgageSuccess` in `mortgage/actions.js`:

https://github.com/decentraland/marketplace/pull/244/files#diff-1bc520535984d0e30212dd309cfb59b1R46

```javascript
  const { x, y } = splitCoordinate(assetId)
// Should be
  const [ x, y ] = splitCoordinate(assetId)
```

File: webapp/src/components/ParcelCard/ParcelCard.css [hunk 6]
```diff
@@ -2,11 +2,18 @@
...
   text-transform: capitalize;
+}
+
+.ParcelCard .Mortgage.pending{
   color: var(--orange);
 }
 
+.ParcelCard .mortgage-status.ongoing{
+  color: var(--light-blue);
+}
```
Comment (NicoSantangelo) [inline webapp/src/components/ParcelCard/ParcelCard.css]: The annoying co-worker strikes again. Can we prettify this file? to look like `.selector {` instead of `.selector{`

File: webapp/src/components/ParcelDetailPage/ParcelDetailPage.container.js [hunk 7]
```diff
@@ -8,28 +8,24 @@ import { getDistricts } from 'modules/districts/selectors'
...
-      isLoading: isLoading(state),
-      error: getError(state),
-      districts: getDistricts(state),
-      publications: getParcelPublications(state),
-      mortgages: getParcelMortgages(state)
-    }
-  }
+  return state => ({
+    x,
+    y,
```
Comment (NicoSantangelo) [inline webapp/src/components/ParcelDetailPage/ParcelDetailPage.container.js]: Warn!

Comment (NicoSantangelo) [inline webapp/src/components/ParcelDetailPage/ParcelDetailPage.container.js]: ⚠️ 

I think this is re-introducing a bug where the `x,y` coordinates won't change and in consequence the detail won't reload when clicking the map

File: webapp/src/components/PayMortgagePage/PayMortgageForm/PayMortgageForm.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        )
+      })
+    }
+  }
+
+  handleSubmit = () => {
+    const {
+      mortgage: { loan_id, asset_id },
+      onSubmit
+    } = this.props
```
Comment (NicoSantangelo) [inline webapp/src/components/PayMortgagePage/PayMortgageForm/PayMortgageForm.js]: _You don't need to change this_: this is perfectly valid, but I lean towards using something like:

```javascript
const { mortgage, onSubmit } = this.props
const { loan_id, asset_id } = morgage
```

it accomplishes the same in fewer lines and it's consistent with the rest of the codebase

File: webapp/src/components/PayMortgagePage/PayMortgagePage.container.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+} from 'modules/mortgage/selectors'
+import { isLoading } from 'modules/parcels/selectors'
+import { getWallet, isConnected, isConnecting } from 'modules/wallet/selectors'
+
+const mapState = (state, ownProps) => {
+  const { x, y } = getMatchParamsCoordinates(ownProps)
+  const getMortgage = getParcelMortgageFactory(x, y)
+  return state => ({
+    x,
+    y,
```
Comment (NicoSantangelo) [inline webapp/src/components/PayMortgagePage/PayMortgagePage.container.js]: Same as `ParcelDetailPage` above, the `x,y` won't change if the reducer updates. Is this a problem here?

File: webapp/src/components/ParcelCard/ParcelCard.css (hunk 6 above)
Comment (NicoSantangelo) [reply webapp/src/components/ParcelCard/ParcelCard.css]: Same goes for the other `.css` files, but then again, it's not really important

File: webapp/src/lib/utils.js [hunk 10]
```diff
@@ -99,26 +99,25 @@ export function isExpired(expires_at) {
...
+ * @param  {array | string} - status
  * @returns {boolean}
  */
 export function hasStatus(obj, status) {
   return (
     obj &&
-    obj.status === status &&
-    obj.tx_status === txUtils.TRANSACTION_STATUS.confirmed &&
-    !isExpired(obj)
+    status.includes(obj.status) &&
```
Comment (NicoSantangelo) [inline webapp/src/lib/utils.js]: This shouldn't matter in the long run, buuut, if you have an status named `open` and another named `opened` then:

```javascript
  hasStatus({ status: 'open' }, 'opened') // true
```

I don't think it's a really a big deal, but wanted to point it out just in case

File: webapp/src/modules/mortgage/utils.js [hunk 11]
```diff
@@ -21,11 +30,39 @@ export function getLoanMetadata() {
...
-      mortgage &&
-      mortgage.borrower === borrower &&
-      mortgage.status !== MORTGAGE_STATUS.cancelled
+export function isMortgageActive(mortgage, parcel) {
+  return (
+    mortgage &&
+    parcel &&
+    ((isMortgagePending(mortgage) &&
+      isOpen(parcel.publication, PUBLICATION_STATUS.open)) ||
+      hasStatus(mortgage, [MORTGAGE_STATUS.ongoing, MORTGAGE_STATUS.paid]))
```
Comment (NicoSantangelo) [inline webapp/src/modules/mortgage/utils.js]: I think this is kind of confusing I can't see what it's doing at first glance. Can we maybe reword it?

Like

```javascript
  if (mortgage && parcel) {
    const isPending = isMortgagePending(mortgage) && isOpen(parcel.publication, PUBLICATION_STATUS.open)
    
    return isPending || hasStatus(mortgage, [MORTGAGE_STATUS.ongoing, MORTGAGE_STATUS.paid])
  }
```

(wrote on Github, don't take seriously)

File: webapp/src/modules/mortgage/utils.js [hunk 12]
```diff
@@ -21,11 +30,39 @@ export function getLoanMetadata() {
...
+}
+
+/**
+ * Returns mortgages active -> pending status require an open publication
+ * @param {array} - mortgages
+ * @param  {array} - parcels
+ * @returns {array} - mortgages
+ */
+export function getActiveMortgages(mortgages, parcels) {
+  return mortgages
```
Comment (NicoSantangelo) [inline webapp/src/modules/mortgage/utils.js]: if mortgages could be undefined, maybe instead of adding a `?:` we ca use:

`export function getActiveMortgages(mortgages = [], parcels = {}) {`

or 

export function getActiveMortgages(mortgages, parcels) {
  mortgages = mortgages || []

Comment (NicoSantangelo) [inline webapp/src/modules/mortgage/utils.js]: if mortgages could be undefined, maybe instead of adding a `?:` we ca use:

`export function getActiveMortgages(mortgages = [], parcels = {}) {`

or 

```javascript
export function getActiveMortgages(mortgages, parcels) {
  mortgages = mortgages || []
```

File: scripts/monitor/processEvents.js (hunk 2 above)
Comment (nachomazzara) [reply scripts/monitor/processEvents.js]: 🤐

File: webapp/src/components/PayMortgagePage/PayMortgagePage.container.js (hunk 9 above)
Comment (nachomazzara) [reply webapp/src/components/PayMortgagePage/PayMortgagePage.container.js]: It should not cause any problem here, no need for re-fetching

File: webapp/src/lib/utils.js (hunk 10 above)
Comment (nachomazzara) [reply webapp/src/lib/utils.js]: Cool, what do you think of refactoring it once we need it?
